src/neo/Cryptography/BloomFilter.cs

```diff
@@ -37,18 +37,33 @@ public class BloomFilter
         /// </summary>
         public uint Tweak { get; private set; }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BloomFilter"/> class.
+        /// </summary>
+        /// <param name="m">The size of the bit array used by the bloom filter.</param>
+        /// <param name="k">The number of hash functions used by the bloom filter.</param>
+        /// <param name="nTweak">Used to generate the seeds of the murmur hash functions.</param>
+        public BloomFilter(int m, int k, uint nTweak)
+        {
+            if (k < 0 || m < 0) throw new ArgumentOutOfRangeException();
+            this.seeds = Enumerable.Range(0, k).Select(p => (uint)p * 0xFBA4C795 + nTweak).ToArray();
+            this.bits = new BitArray(m);
+            this.bits.Length = m;
+            this.Tweak = nTweak;
+        }
+
         /// <summary>
         /// Initializes a new instance of the <see cref="BloomFilter"/> class.
         /// </summary>
         /// <param name="m">The size of the bit array used by the bloom filter.</param>
         /// <param name="k">The number of hash functions used by the bloom filter.</param>
         /// <param name="nTweak">Used to generate the seeds of the murmur hash functions.</param>
         /// <param name="elements">The initial elements contained in this <see cref="BloomFilter"/> object.</param>
-        public BloomFilter(int m, int k, uint nTweak, byte[] elements = null)
+        public BloomFilter(int m, int k, uint nTweak, ReadOnlyMemory<byte> elements)
```

Comment:
This change must be reverted. Because you're calling `elements.ToArray()` below, and it lose performance.

Reply:
Before the change we call ToArray inside the Payload, and only one bloom filter is created per payload, so the `ToArray` quantity is the same, but if you don't create the bloom filter, you can save one.

Reply:
OK.